Tests/Tests/STPPaymentMethodsViewControllerTest.m

```diff
@@ -0,0 +1,237 @@
+//
+//  STPPaymentMethodsViewControllerTest.m
+//  Stripe
+//
+//  Created by Ben Guo on 4/10/17.
+//  Copyright © 2017 Stripe, Inc. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import <OCMock/OCMock.h>
+#import "STPFixtures.h"
+#import "STPMocks.h"
+#import "STPPaymentMethodsInternalViewController.h"
+
+@interface STPPaymentMethodsViewController (Testing)
+@property(nonatomic, weak)UIViewController *internalViewController;
+@end
+
+@interface STPPaymentMethodsViewControllerTest : XCTestCase
+
+@end
+
+@implementation STPPaymentMethodsViewControllerTest
+
+- (STPPaymentMethodsViewController *)buildViewControllerWithCustomer:(STPCustomer *)customer
+                                                       configuration:(STPPaymentConfiguration *)config
+                                                            delegate:(id<STPPaymentMethodsViewControllerDelegate>)delegate {
+    STPTheme *theme = [STPTheme defaultTheme];
+    id<STPBackendAPIAdapter> mockAPIAdapter = [STPMocks staticAPIAdapterWithCustomer:customer];
+    STPPaymentMethodsViewController *vc = [[STPPaymentMethodsViewController alloc] initWithConfiguration:config
+                                                                                                   theme:theme
+                                                                                              apiAdapter:mockAPIAdapter
+                                                                                                delegate:delegate];
+    if (vc) {
+        XCTAssertNotNil(vc.view);
+    }
+    return vc;
+}
+
+/**
+ When the customer has no sources and the configuration doesn't use sources,
+ STPAddCardViewController should be shown.
+ */
+- (void)testInitWithNoSourcesAndUseSourcesOff {
+    STPCustomer *customer = [STPFixtures customerWithNoSources];
+    STPPaymentConfiguration *config = [STPFixtures paymentConfiguration];
+    config.useSourcesForCards = NO;
+    id<STPPaymentMethodsViewControllerDelegate>delegate = OCMProtocolMock(@protocol(STPPaymentMethodsViewControllerDelegate));
+    STPPaymentMethodsViewController *sut = [self buildViewControllerWithCustomer:customer
+                                                                   configuration:config
+                                                                        delegate:delegate];
+    XCTAssertTrue([sut.internalViewController isKindOfClass:[STPAddCardViewController class]]);
+}
+
+/**
+ When the customer has no sources and the configuration uses sources,
+ STPAddSourceViewController should be shown.
+ */
+- (void)testInitWithNoSourcesAndUseSourcesOn {
```

Comment:
ah yeah, that's a good point – I'll explicitly set up availablePaymentMethods in the config, and add some additional cases.